### PR TITLE
[Textual] Added the 'prettyprinter-configurable' library

### PR DESCRIPTION
**Pull request by effectfully**
Do not review yet, that's just for CI and self-review.

### CODE REVIEW

File: language-plutus-core/test/Evaluation/Machines/Budget/Examples/InterList/FoldrInterList.plc.golden [hunk 1]
```diff
@@ -1,4 +1,4 @@
-((Left An error has occurred:  User error:
+( (Left An error has occurred:  User error:
```
Comment (effectfully) [inline language-plutus-core/test/Evaluation/Machines/Budget/Examples/InterList/FoldrInterList.plc.golden:1]: The `PrettyBy config (a, b)` instance previously defined in `PlutusPrelude` wasn't behaving the same way the `Pretty (a, b)` instance behaves. I've fixed that, hence the changes.

**Comment by effectfully** [comment]
Ok, should be ready for review.

I'll think if more docs are needed, there's one thing I want to tweak and I still need to resolve the build problems.

**Comment by michaelpj** [comment]
I really don't want a nested stack project. I'm willing to fight the Nix stuff to make it work as it is.

**Comment by effectfully** [comment]
> I really don't want a nested stack project. I'm willing to fight the Nix stuff to make it work as a normal local package.

I was going to experiment with that and if it works, then propose to structure things this way. But if you don't want that, we can skip the experimentation part.

There's still another option: create a separate repo (after getting an approval for that).

**Comment by michaelpj** [comment]
Needs a rebase, I think.

**Pull request merged**